packages/frameworks-qwik/src/index.ts

```diff
@@ -0,0 +1,388 @@
+    if (isServer) {
+      const prefix: string = "/api/auth"
+
+      const action = req.url.pathname
+        .slice(prefix.length + 1)
+        .split("/")[0] as AuthAction
+
+      const auth = await patchAuthOptions(authOptions, req)
+      if (
+        qwikAuthActions.includes(action) &&
+        req.url.pathname.startsWith(prefix + "/")
+      ) {
+        const res = (await Auth(req.request, auth)) as Response
+        const cookie = res.headers.get("set-cookie")
+        if (cookie) {
+          req.headers.set("set-cookie", cookie)
+          res.headers.delete("set-cookie")
+          fixCookies(req)
+        }
+        throw req.send(res)
+      } else {
+        const { data, cookie } = await getSessionData(req.request, auth)
+        req.sharedMap.set("session", data)
+        if (cookie) {
+          req.headers.set("set-cookie", cookie)
+          fixCookies(req)
+        }
+      }
+    }
+  }
+
+  return {
+    useAuthSignIn,
+    useAuthSignOut,
+    useAuthSession,
+    onRequest,
+  }
+}
+
+export const qwikAuth$ = /*#__PURE__*/ implicit$FirstArg(qwikAuthQrl)
+
+async function authAction(
+  body: URLSearchParams | undefined,
+  req: RequestEventCommon,
+  path: string,
+  authOptions: QwikAuthConfig
+) {
+  const request = new Request(new URL(path, req.request.url), {
+    method: req.request.method,
+    headers: req.request.headers,
+    body: body,
+  })
+  request.headers.set("content-type", "application/x-www-form-urlencoded")
+  const res = (await Auth(request, {
+    ...authOptions,
+    skipCSRFCheck,
+  })) as Response
+
+  const cookies: string[] = []
+  res.headers.forEach((value, key) => {
+    if (key === "set-cookie") {
+      // while browsers would support setting multiple cookies, the fetch implementation does not, so we join them later.
+      cookies.push(value)
+    } else if (!req.headers.has(key)) {
+      req.headers.set(key, value)
+    }
+  })
+
+  if (cookies.length > 0) {
+    req.headers.set("set-cookie", cookies.join(", "))
+  }
+
+  fixCookies(req)
+
+  try {
+    return await res.json()
+  } catch (error) {
+    return await res.text()
+  }
+}
+
+const fixCookies = (req: RequestEventCommon) => {
+  req.headers.set("set-cookie", req.headers.get("set-cookie") || "")
+  const cookie = req.headers.get("set-cookie")
+  if (cookie) {
+    req.headers.delete("set-cookie")
+    splitCookiesString(cookie).forEach((cookie) => {
+      const { name, value, ...rest } = parseString(cookie)
+      req.cookie.set(name, value, rest as any)
+    })
+  }
+}
+
+export const ensureAuthMiddleware = (req: RequestEvent) => {
+  const isLoggedIn = req.sharedMap.has("session")
+  if (!isLoggedIn) {
+    throw req.error(403, "sfs")
+  }
+}
+
+const defaultCallbackURL = (req: RequestEventCommon) => {
+  req.url.searchParams.delete("qaction")
+  return req.url.href
+}
+
+async function getSessionData(
+  req: Request,
+  options: AuthConfig
+): GetSessionResult {
+  options.secret ??= process.env.AUTH_SECRET
```

Comment:
I think we should call `fillOptionsWithEnvVariables` here

Reply:
options is already patched here. I changed the name of the variable and clean up the code. Thanks